test-result-summary-client/src/Dashboard/Widgets/Graph/ODM.jsx

```diff
@@ -138,23 +146,33 @@ export default class ODM extends Component {
         this.setState( { displaySeries } );
     }
 
+
     formatter = function() {
-        const x = new Date( this.x );
+        const x = this.x;
         if ( this.point.additionalData ) {
             let buildLinks = '';
             let i = this.series.data.indexOf(this.point);
             let prevPoint = i === 0 ? null : this.series.data[i - 1];
             this.point.additionalData.forEach(( xy, i ) => {
-                const { testId, buildName, buildNum } = xy;
-                buildLinks = buildLinks + ` <a href="/output/test?id=${testId}">${buildName} #${buildNum}</a>`;
+                const { testId, parentId, buildName, buildNum, hasChildren } = xy;
+                let childBuildLinks = '' + ` <a href="/output/test?id=${testId}">${buildName} #${buildNum}</a>`;
+                let parentBuildLinks = '' + ` <a href="/buildDetail?parentId=${parentId}">${buildName} #${buildNum}</a>`
+                buildLinks = hasChildren ? parentBuildLinks : childBuildLinks;
             } );
 
             let lengthThis = this.point.additionalData.length;
             let lengthPrev = prevPoint ? prevPoint.additionalData.length : 0;
 
             let javaVersion = this.point.additionalData[lengthThis - 1].javaVersion;
             let prevJavaVersion = prevPoint ? prevPoint.additionalData[lengthPrev - 1].javaVersion : null;
-            let ret = `${this.series.name}: ${this.y}<br/> Build: ${x.toISOString().slice( 0, 10 )} <br/>Link to builds: ${buildLinks}`;
+            let mean = this.point.additionalData[lengthThis - 1].mean;
+            let max = this.point.additionalData[lengthThis - 1].max;
+            let min = this.point.additionalData[lengthThis - 1].min;
+            let median = this.point.additionalData[lengthThis - 1].median;
+            let stddev = this.point.additionalData[lengthThis - 1].stddev;
+            let CI = this.point.additionalData[lengthThis - 1].CI;
+            let validIterations = this.point.additionalData[lengthThis - 1].validIterations;
+            let ret = `${this.series.name}: ${math.round(this.y, 3)}<br/> Build: ${x} <br/>Link to builds: ${ buildLinks } <br/><b>Aggregate Infomation:</b><br/>mean: ${mean}<br/>max: ${max}<br/>min: ${min}<br/>median: ${median}<br/>stddev: ${stddev}<br/>CI: ${CI}<br/>validIterations: ${validIterations}`;

```

Comment:
Typo: `Infomation`! Let's just remove `Aggregate Information` string to save space.

Reply:
Instead of using `${this.series.name}`, we should call it `Test vs Baseline`. We also need to add `%` after `${math.round(this.y, 3)}`.